### DISCUSSION TITLE
v2 Alpha Testing - Round 2

### DISCUSSION THREAD

**Discussion by jwildfire** [SafetyGraphics/safetyGraphics#605] [General]
# Summary

The goal of this thread is to facilitate a second round of alpha testing on the v2.0 release of the safetyGraphics app. This builds upon the first round of alpha testing discussed [here](#538). Alpha testers are being asked to complete 2 tasks: 

1. Provide feedback on the updated functionality in the [default app](https://jwildfire.shinyapps.io/safetyGraphics/).
1. Provide feedback on the new `safetyGraphicsInit()`.

See the following sections for more details. 

# Part 1 - Provide feedback on the updates in the default app

Access the app with [this link](https://safety-graphics.shinyapps.io/safetyGraphics/) or install the package using the instructions below and then initialize the default app by calling `safetyGraphicsApp()`.  Any feedback is welcome, but place a special emphasis on reviewing the updated chart export process and settings tab, which have both been updated since the first Alpha testing. 

# Part 2 - Try the new app initialization workflow. 

Install the package  using the instructions below and then load the chart initialization app by calling `safetyGraphicsInit()`. Try loading data for multiple domains using the UI, selecting a custom set of charts and then running the app. 

# Installation Instructions

Insure that your are using R version 4 or higher, and then run the following code to install the app: 

``` 
install.packages("devtools")
library("devtools")
devtools::install_github("safetyGraphics/safetyGraphics", ref="v2.0.0-alpha2") 
library(safetyGraphics)
devtools::install_github("safetyGraphics/safetyCharts", ref="v1.0.0-alpha2") 
library(safetyCharts)
```

`devtools` will likely prompt you to install/update some packages, say yes!

**Comment by jwildfire** [comment]
@samussiah @xni7 @elimillera @JimBuchanan @mli1 - I think we're ready for a quick second round of alpha testing! Just let me know if you have any questions and we can discuss here!

**Comment by JimBuchanan** [comment]
Here are my findings so far. Because I took some screenshots, I've attached a Word document with my collective observations.

[Alpha testing - Jim.docx](https://github.com/SafetyGraphics/safetyGraphics/files/6969856/Alpha.testing.-.Jim.docx)

**Reply by jwildfire** [reply]
Thanks for the nice write-up, @JimBuchanan! Will reply in more detail later, but first a quick question - How were you using the app? Default or custom data? Did you use the init workflow here? Most of the comments are expected, but I'm a bit confused about why the paneled outlier explorer isn't running properly ...

**Reply by jwildfire** [reply]
![image](https://user-images.githubusercontent.com/3680095/129074163-77b22ead-73a7-49d5-a219-b7c2763e2d41.png)
This error makes me think you've got some non-numeric data in your visit number column (which could be causing the issues in outlier explorer and paneled outlier explorer)

**Reply by JimBuchanan** [reply]
It looks like the VISITNUM column has text formatted as "General". It contains terms such as "Day 1", "Week 01", "Week 02", etc.

**Comment by JimBuchanan** [comment]
Hi Jeremy,

I ran the default code in RStudio and then ran safetyGraphicsInit(). From there I loaded two sets of lab data. One was a set designed for the hepatic explorer. The second was a set I used for the laboratory analysis exercise I did for Abbvie. It contains a broader range of lab analytes which I ran with the (previous) outlier explorer. I’ve attached the dataset if that helps.

Thanks,

Jim

**Comment by JimBuchanan** [comment]
In the Hepatic Explorer, the R and nR values over time are displayed along with the ALT, AST, total bilirubin and alkaline phosphatase value in the section Standardized Lab Values by Study Day. Previously, the R ratio over time was displayed in the section R Ratio by Study Day. That section no longer has a display (because it's not needed now), but the section heading still remains. We can remove the R Ratio by Study Day section heading.
